corehq/apps/api/es.py

```diff
@@ -701,35 +513,55 @@ def consume_params(self, raw_params):
 ]
 
 
-def es_search(request, domain, reserved_query_params=None):
-    return es_search_by_params(request.GET, domain, reserved_query_params)
+def _validate_and_get_es_filter(search_param):
+    #
+    try:
+        _filter = search_param.pop('filter')
+        # custom use case by 'enveritas' project for Form API
+        date_range = _filter['range']['inserted_at']
+        return {
+            'range': {'inserted_at': date_range}
+        }
+    except KeyError:
```

Comment:
if this is raised then `_filter` may not exists below. I suggest popping it with a default of None outside of the try block and then testing if the it is not None